src/shadowbox/infrastructure/logging.ts

```diff
@@ -12,25 +12,45 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import * as path from 'path';
 
-function makeLogMessage(level: string, message: string): string {
+interface Callsite {
+  getLineNumber(): number;
+  getFileName(): string;
+}
+
+function getCallsite(): Callsite {
+  const originalPrepareStackTrace = Error.prepareStackTrace;
+  Error.prepareStackTrace = (_, stack) => {
+      return stack;
+  };
+  const error = new Error();
+  Error.captureStackTrace(error, getCallsite);
+  // tslint:disable-next-line:no-any
+  const stack = error.stack as any as Callsite[];
+  Error.prepareStackTrace = originalPrepareStackTrace;
+  return stack[1];
+}
+
+function makeLogMessage(level: string, callsite: Callsite, message: string): string {
   // This creates a string in the UTC timezone
   // See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString
-  return `${level}:${new Date().toISOString()}] ${message}`;
+  const timeStr = new Date().toISOString();
+  return `${level}${timeStr} ${process.pid} ${path.basename(callsite.getFileName())}:${callsite.getLineNumber()}] ${message}`;
 }
 
 export function error(message: string) {
-  console.error(makeLogMessage('E', message));
+  console.error(makeLogMessage('E', getCallsite(), message));
```

Comment:
Seems like the call to `getCallsite()` can be pushed down into `makeLogMessage`.

Reply:
Not really, since then I'll get line from logging.ts. Unless I pass some parameter indicating how many levels to pop, but then I might as well pass the callsite.

Reply:
Seems like you are already getting lines you don't want, then - but I don't feel strongly about it.